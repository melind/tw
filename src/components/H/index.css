

.H {
  background-color: white;
  height: 80%;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #d6b519;
}

.H img {
  width: 10%;
}