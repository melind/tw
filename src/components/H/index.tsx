import React, {useState} from 'react';
import {Link} from 'react-router-dom';
import Greeting from '../Greeting';
import './index.css';

const H = () => {

    const [loading, setLoading] = useState(true);
    
    setTimeout(function () {
     setLoading(false);
   }, 3000);
   if (loading) {
          return <Greeting />
        }

return (
       <div className="H"> 
                      
                     <h1>Bienvenue </h1> 
                     < img src="../../../images/bobine-noir-blanc.png" alt="bobine de film" />
                     <Link to="/signup">SignUp</Link>
                     < img src="../../../images/bobine.png" alt="bobine de film"/>
                     <Link to="/login">login</Link><br/>
                     <div><a target="_blank" href="/icons/set/film-reel" rel="noopener noreferrer">Bobine de film icon</a> by <a target="_blank" href="https://icones8.fr" rel="noopener noreferrer">Icons8</a></div>
                    </div>
    )
    
} 

export default H;