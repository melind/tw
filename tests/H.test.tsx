import React from 'react';
import { shallow, mount, render} from 'enzyme';
import {Link} from 'react-router-dom';
//import ReactDOM from 'react-dom';
import H from '../src/components/H';
/** test to check that we don't broke our code in the evolution */


/* snapshot tool let us to follow the evlution of our H,
 it compare the dom of our component save in file snap and the actual dom*/

/* test de rendu*/
describe('h', () => {
it('h should render correctly', () => {
  const h = shallow(<H ><div > 
                     <header > 
                     <h1>Accueil des non connectés </h1> 
                     <Link to="/signup">SignUp</Link>
                     <Link to="/login">login</Link>
                     </header>
                    </div></H>);//use render() of component
  expect(h).toMatchSnapshot();
});
/* test de fonctionnalité*/
 it('renders h', () => {
     const wrapper = shallow(<H optInValue={true} authenticated={true}/>);

     expect(wrapper.length).toEqual(1);   
  }); 

  it('h contains', () => {
    const wrapper = shallow((<header > 
                     <h1>Accueil des non connectés </h1>
                     </header>));
        expect(wrapper.contains(<h1>Accueil des non connectés </h1>)).toEqual(true);
  });

});
