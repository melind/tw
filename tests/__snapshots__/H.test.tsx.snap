// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h h should render correctly 1`] = `ShallowWrapper {}`;
